remote/mdb_io.go

```diff
@@ -283,3 +284,88 @@ type loggingCacheLenRequest struct {
 type loggingSendMessageRequest struct {
 	Payload options.LoggingPayload `bson:"send_message"`
 }
+
+type scriptingCreateRequest struct {
+	Params struct {
+		Type    string `bson:"type"`
+		Options []byte `bson:"options"`
+	} `bson:"create_scripting"`
+}
+
+type scriptingCreateResponse struct {
+	shell.ErrorResponse `bson:"error_response,inline"`
+	ID                  string `bson:"id"`
+}
+
+func makeScriptingCreateResponse(id string) *scriptingCreateResponse {
+	return &scriptingCreateResponse{
+		ErrorResponse: shell.MakeSuccessResponse(),
+		ID:            id,
+	}
+}
+
+type scriptingGetRequest struct {
+	ID string `bson:"get_scripting"`
+}
+
+type scriptingSetupRequest struct {
+	ID string `bson:"setup_scripting"`
+}
+
+type scriptingCleanupRequest struct {
+	ID string `bson:"cleanup_scripting"`
+}
+
+type scriptingRunRequest struct {
+	Params struct {
+		ID   string   `bson:"id"`
+		Args []string `bson:"args"`
+	} `bson:"run_scripting"`
+}
+
+type scriptingRunScriptRequest struct {
+	Params struct {
+		ID     string `bson:"id"`
+		Script string `bson:"script"`
+	} `bson:"run_script_scripting"`
+}
+
+type scriptingBuildRequest struct {
+	Params struct {
+		ID   string   `bson:"id"`
+		Dir  string   `bson:"dir"`
+		Args []string `bson:"args"`
+	} `bson:"build_scripting"`
+}
+
+type scriptingBuildResponse struct {
+	shell.ErrorResponse `bson:"error_response,inline"`
```

Comment:
are we using bson parsers that respect inline?

Reply:
Yes, inline is respected here.